securedrop/source_app/utils.py

```diff
@@ -8,18 +8,20 @@
 from sqlalchemy.orm import sessionmaker
 from threading import Thread
 
+from typing import Optional
+
 import i18n
 import re
 
 from crypto_util import CryptoException
 from models import Source
+from sdconfig import SDConfig
 
-
-def logged_in():
+def logged_in() -> bool:
     return 'logged_in' in session
 
 
-def valid_codename(codename):
+def valid_codename(codename: str) -> bool:
```

Comment:
Should this be `-> Optional[bool]`?

Reply:
`bool` is right since you'll either return a boolean _or_ it'll raise an exception (what Flask/Werkzeug `abort` is doing, which looks to be annotated properly as `typing.NoReturn` [here](https://github.com/python/typeshed/blob/master/third_party/2and3/werkzeug/exceptions.pyi#L184))